lighthouse-core/audits/seo/hreflang.js

```diff
@@ -76,33 +88,62 @@ class Hreflang extends Audit {
    * @param {LH.Artifacts} artifacts
    * @return {LH.Audit.Product}
    */
-  static audit(artifacts) {
-    /** @type {Array<{source: string|{type: 'node', snippet: string}}>} */
+  static audit({LinkElements}) {
+    /** @type {Array<{source: string|{type: 'node', snippet: string}, reason: string}>} */
     const invalidHreflangs = [];
 
-    for (const link of artifacts.LinkElements) {
-      if (link.rel !== 'alternate') continue;
-      if (!link.hreflang || isValidHreflang(link.hreflang)) continue;
-      if (link.source === 'body') continue;
+    const auditableLinkElements = LinkElements.filter(linkElement => {
+      const isAlternate = linkElement.rel === 'alternate';
+      const hasHreflang = linkElement.hreflang;
+      const isInBody = linkElement.source === 'body';
+
+      return isAlternate && hasHreflang && !isInBody;
+    });
+
+    for (const link of auditableLinkElements) {
+      const reasons = [];
+      let source;
```

Comment:
I think this is what you're intending here?

```suggestion
      /** @type {string|LH.Audit.Details.NodeValue|undefined} */
      let source;
```

Reply:
Thank you!